src/SIP2Client.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace NYPL\Services;
+
+use NYPL\Starter\APIException;
+use NYPL\Starter\Config;
+use PHPUnit\Framework\Exception;
+use sip2;
+
+class SIP2Client
+{
+    /**
+     * @var \sip2
+     */
+    protected $sip2Client;
+
+    public function __construct()
+    {
+        $this->initializeSip2Client();
+    }
+
+    /**
+     * @throws APIException
+     * @return \sip2
+     */
+    public function getSip2Client()
+    {
+        return $this->sip2Client;
+    }
+
+    /**
+     * @param \sip2 $sip2Client
+     */
+    public function setSip2Client(\sip2 $sip2Client)
+    {
+        $this->sip2Client = $sip2Client;
+    }
+
+    /**
+     * @throws APIException
+     * @return \sip2
+     */
+    protected function initializeSip2Client()
+    {
+        $sipClient = new sip2();
+
+        try {
+            $sipClient->hostname = Config::get('SIP2_HOSTNAME');
+            $sipClient->port = Config::get('SIP2_PORT');
+
+            $sipClient->connect();
+
+            $sipClient->AC = Config::get('SIP2_TERMINAL_PASSWORD', null, true);
```

Comment:
What does AC mean?

Reply:
This abbreviation along with others, AO, AB, AQ, AJ, AA, AD, etc., are SIP2 protocol IDs that they define in their specification. See https://developers.exlibrisgroup.com/alma/integrations/selfcheck/sip2

AC is the ID for the terminal password used by SIP2 in formulating its message.

Reply:
Roger, approving.